src/packages/solid/withScrolling.ts

```diff
@@ -0,0 +1,88 @@
+export function withScrolling(direction: 'column' | 'row', adjustment: number = 0) {
+  return (componentRef, selectedElement, selected, lastSelected) => {
+    let previousVal, newVal, next, size;
+    const gap = componentRef.gap || 0;
+    const scrollType = componentRef.scrollType;
+    const [lastItem, windowVal] = updateLastIndex(componentRef, direction);
+
+    if (componentRef.children.length === 0) {
+      return;
+    }
+
+    // values based on row or column
+    if (direction === 'row') {
+      previousVal = componentRef.x;
+      newVal = selectedElement.x;
+      size = selectedElement.width;
+    } else {
+      previousVal = componentRef.y;
+      newVal = selectedElement.y;
+      size = selectedElement.height;
+    }
```

Comment:
since these values aren't being mutated at any point, can we instead set them as constants using ternaries?

```
const previousVal = direction === 'row' ? componentRef.x : componentRef.y;
const newVal = direction === 'row' ? selectedElement.x : selectedElement.y;
const size = direction === 'row' ? selectedElement.width : selectedElement.height;
```